### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by goldenuni**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
+
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
```
Comment (Bohdan-Salamakha) [inline app/main.py:11]: add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contain string elements has typehint list[str]

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

File: app/main.py [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
+
+
+def create_person_list(people: list[dict]) -> list:
```
Comment (Bohdan-Salamakha) [inline app/main.py:11]: add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contain string elements has typehint list[str]

for return typehint too

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v2

File: app/main.py [hunk 3]
```diff
@@ -1,8 +1,20 @@
...
+        self.age = age
+        Person.people[name] = self
+
+
+def create_person_list(people: list[dict]) -> list[Person]:
+    person_list = []
+    for human in people:
+        person_list.append(Person(human.get("name"), human.get("age")))
+    for i, human in enumerate(people):
+        if "wife" in human and human.get("wife") is not None:
```
Comment (olena-hul) [inline app/main.py:16]:
```suggestion
        if human.get("wife"):
```

File: app/main.py [hunk 4]
```diff
@@ -1,8 +1,20 @@
...
+            person_list[i].wife = Person.people.get(human["wife"])
+        if "husband" in human and human.get("husband") is not None:
```
Comment (olena-hul) [inline app/main.py:18]:
```suggestion
        if human.get("husband"):
```

**Review by olena-hul — CHANGES_REQUESTED** [review]

**Review by olena-hul — APPROVED** [review]

**Comment by olena-hul** [comment]
Good job!

**Pull request closed without merging**